src/google/protobuf/compiler/importer.cc

```diff
@@ -290,7 +290,7 @@ static std::string CanonicalizePath(std::string path) {
   std::vector<std::string> canonical_parts;
   std::vector<std::string> parts = Split(
       path, "/", true);  // Note:  Removes empty parts.
-  for (int i = 0; i < parts.size(); i++) {
+  for (unsigned int i = 0; i < parts.size(); i++) {
```

Comment:
Our style guide is very particular about [integer types](https://google.github.io/styleguide/cppguide.html#Integer_Types), and doesn't allow any of the plain builtin C integer types except for plain `int` (so `unsigned` isn't allowed). I think the best solution for this kind of warning is to change these loops to be range-based for loops.

Reply:
size_t is allowed, although ranged based for loops are usually nicer

Reply:
I fixed some of them with ranges, and reverted some of them (I haven't written any C++ in ages, I went for the easy way) - those warnings are not enabled by default when building with -Werror anyway, so it's not as blocking as the others.